src/validation.cpp

```diff
@@ -2026,6 +2027,33 @@ void UpdateCoins(const CTransaction& tx, CCoinsViewCache& inputs, CTxUndo &txund
     AddCoins(inputs, tx, nHeight);
 }
 
+void UpdateCoinsWithHints(const CTransaction& tx, CCoinsViewCache& inputs, const CBlockIndex& pindex, swiftsync::Aggregate& agg, swiftsync::BlockHints& hints)
+{
+    const bool is_coinbase = tx.IsCoinBase();
+    if (is_coinbase && IsBIP30Unspendable(pindex.GetBlockHash(), pindex.nHeight)) {
+        for (uint64_t index = 0; index < tx.vout.size(); index++) {
+            // No-op on the aggregator and TXO set. These coins will show up again.
+            hints.NextOutput();
+        }
+        return;
+    }
+    if (!is_coinbase) {
+        for (const CTxIn& txin : tx.vin) {
+            agg.Spend(txin.prevout);
+        }
+    }
+    const Txid& txid{tx.GetHash()};
+    for (uint64_t index = 0; index < tx.vout.size(); index++) {
+        COutPoint outpoint = COutPoint(txid, index);
+        if (!hints.IsCurrOutputUnspent() && !tx.vout[index].scriptPubKey.IsUnspendable()) {
```

Comment:
why are we calling `IsUnspendable` just to return immediately in `AddCoin`? We could skip these at the beginning of the loop for both branches.